cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -431,17 +432,15 @@ def _link_rna_sample_to_dna_sample(
             subject_id=rna_sample.subject_id,
             is_tumour=rna_sample.is_tumour,
         )
-        nr_of_subject_id_dna_samples: int = len(subject_id_samples.all())
-        if nr_of_subject_id_dna_samples != 2:
+        subject_id_dna_samples = self._get_application_prep_category(subject_id_samples.all())
+        nr_of_subject_id_dna_samples: int = len(subject_id_dna_samples)
+        if nr_of_subject_id_dna_samples != 1:
```

Comment:
I think you can simplify this
```suggestion
        len(subject_id_dna_samples) != 1:
```